types/src/transaction/transaction.rs

```diff
@@ -295,10 +319,63 @@ impl Transaction {
                 }
                 Ok(())
             }
+            Transaction::AssetUnwrapCCC {
+                burn,
+                ..
+            } => {
+                if burn.prev_out.amount.is_zero() {
+                    return Err(Error::ZeroAmount)
+                }
+                if !burn.prev_out.asset_type.ends_with(&[0; 28]) {
+                    return Err(Error::InvalidAssetType(burn.prev_out.asset_type))
+                }
+                Ok(())
+            }
+        }
+    }
+
+    pub fn unwrapped_amount(&self) -> U256 {
+        match self {
+            Transaction::AssetUnwrapCCC {
+                burn,
+                ..
+            } => burn.prev_out.amount,
+            _ => U256::zero(),
         }
     }
 }
 
+impl InnerTransaction {
+    pub fn hash(&self) -> H256 {
+        match self {
+            InnerTransaction::General(transaction) => transaction.hash(),
+            InnerTransaction::AssetWrapCCC {
+                parcel_hash,
+                ..
+            } => *parcel_hash,
+        }
+    }
+
+    pub fn verify(&self) -> Result<(), Error> {
+        match self {
+            InnerTransaction::General(transaction) => transaction.verify(),
+            InnerTransaction::AssetWrapCCC {
+                output,
+                ..
+            } => match output.amount {
+                Some(amount) if amount.is_zero() => Err(Error::ZeroAmount),
```

Comment:
What happens if the `amount` is `None`?

Reply:
It happened because I just tried to use `AssetMintOutput` as the output format. It might be good to make a type for it...

Reply:
I've changed the type to `AssetWrapCCCOutput`, newly defined one.